ledgers/ccf/transaction_processor/pdo_tp.cpp

```diff
@@ -54,26 +55,67 @@ namespace ccfapp
         };
 
         //======================================================================================================
-        // register enclave expected measurements (member method)
-        auto set_contract_enclave_attestatation_verification_policy = [this](auto& ctx, const nlohmann::json& params) {
+        // register contract enclave attestation check flag (member method)
+        auto set_contract_enclave_attestation_check_flag = [this](auto& ctx, const nlohmann::json& params) {
 
-            const auto in = params.get<RegisterContractEnclaveAttestationVerificationPolicy::In>();
+            const auto in = params.get<RegisterContractEnclaveAttestionCheckFlag::In>();
 
-            // get the current measurements view
-            auto attestation_policy_view = ctx.tx.rw(attestation_policy_table);
+            // get the current view of contract_enclave_check_attestation_flag
+            auto check_attestation_flag_view = ctx.tx.rw(contract_enclave_check_attestation_flag);
+
+            // Current PDO policy permits the flag to be set only once. Check if already set.
+
+            // Below we check the ccf node has a local copy of the attestation flag. If yes, an error
+            // is returned. (Note that global commit of the flag might be pending, and this is OK).
+            auto check_attestation_flag_check = check_attestation_flag_view->get(PDO_ENCLAVE_CHECK_ATTESTATION_FLAG);
+            if (check_attestation_flag_check.has_value()){
+                return ccf::make_success("Attesation check flag can be set only once");
```

Comment:
Shouldn't this be `make_error`? (besides, it says that an error is returned)

Reply:
thats right will fix